vcn.tf

```diff
@@ -4,31 +4,42 @@
 resource "oci_core_vcn" "vcn" {
   # We still allow module users to declare a cidr using `vcn_cidr` instead of the now recommended `vcn_cidrs`, but internally we map both to `cidr_blocks`
   # The module always use the new list of string structure and let the customer update his module definition block at his own pace.
-  cidr_blocks    = var.vcn_cidrs[*]
-  compartment_id = var.compartment_id
-  display_name   = var.label_prefix == "none" ? var.vcn_name : "${var.label_prefix}-${var.vcn_name}"
-  dns_label      = var.vcn_dns_label
-  is_ipv6enabled = var.enable_ipv6
+  cidr_blocks                      = var.vcn_cidrs[*]
+  compartment_id                   = var.compartment_id
+  display_name                     = var.label_prefix == "none" ? var.vcn_name : "${var.label_prefix}-${var.vcn_name}"
+  dns_label                        = var.vcn_dns_label
+  is_ipv6enabled                   = var.enable_ipv6
+  is_oracle_gua_allocation_enabled = var.enable_ipv6 ? var.vcn_is_oracle_gua_allocation_enabled : null
+  ipv6private_cidr_blocks          = var.enable_ipv6 ? var.vcn_ipv6private_cidr_blocks : null

```

Comment:
`ipv6private_cidr_blocks` is a crappy and thoughtless variable name and is not consistent with what vcn uses for ipv4 (not your fault) in the oci provider.

I'm willing to make a breaking change and release a new major version so we can handle this better:

- instead of var.vcn_cidrs, we can use vcn_cidrs4 or vcn_cidrs_ipv4 
- instead of var.vcn_ipv6private_cidr_blocks, we can use vcn_cidrs6 or vcn_cidrs_ipv6

Or something along that line i.e. don't make the parameters gouge the eyes of developers who are going to use them. 

Since we are thinking of making a breaking change, consider renaming other variables accordingly.